beacon-chain/endtoend/evaluators.go

```diff
@@ -0,0 +1,144 @@
+package endtoend
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	ptypes "github.com/gogo/protobuf/types"
+	"github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+// policy defines an function that returns a bool
+// for whether the evaluator it is assigned to runs.
+type policy func(currentEpoch uint64) bool
+
+// evaluation defines a function that takes in BeaconChainClient
+// and uses it to monitor the chain, to determine where it fulfills the conditions required.
+type evaluation func(client eth.BeaconChainClient) error
+
+// evaluator defines a struct for executing a given evaluation
+// on a running E2E test given that the policy returns true.
+type evaluator struct {
+	name       string
+	policy     policy
+	evaluation evaluation
+}
+
+// runEvaluators takes in testing.T, BeaconChainClient and an array of evaluators.
+// Using the client to check the state, it runs each evaluator in a subtest.
+func runEvaluators(t *testing.T, client eth.BeaconChainClient, evaluators []evaluator) {
+	in := new(ptypes.Empty)
+	chainHead, err := client.GetChainHead(context.Background(), in)
+	if err != nil {
+		t.Errorf("failed to get chain head: %v", err)
+	}
+	currentEpoch := chainHead.BlockSlot / params.BeaconConfig().SlotsPerEpoch
+
+	for _, evaluator := range evaluators {
+		if evaluator.policy(currentEpoch) {
+			fmt.Printf("Running %s\n", evaluator.name)
```

Comment:
Use `t.Logf`

Reply:
Do not use fmt.Printf. You don't need this print statement at all because go prints the test name when it runs.